mfs/system.go

```diff
@@ -123,6 +123,36 @@ func (kr *Root) Flush() error {
 	return nil
 }
 
+// FlushMemFree flushes the root directory and then uncaches all of its links.
+// This has the effect of clearing out potentially stale references and allows
+// them to be garbage collected.
+// CAUTION: Take care not to ever call this while holding a reference to any
+// child directories. Those directories will be bad references and using them
+// may have unintended racy side effects.
```

Comment:
What exactly do you mean here? How will removing these from the map lead to race conditions?

I think what's throwing me is the term "cache". Is it a write cache? A read cache?